woocommerce/woocommerce
# Filtering for the COT admin list table.

Implements filtering (by date, by customer) for the COT admin list table.

![month-customer-filtering-cot](https://user-images.githubusercontent.com/3594411/178067746-3b8576b8-7d87-4ae1-ac56-942de981d400.gif)

Closes #33196.

### How to test the changes in this Pull Request:

Setup:

1. Enable the COT feature.
2. Select **Use the WooCommerce orders tables** from **WooCommerce ▸ Settings ▸ Advanced ▸ Custom Data Stores**.
3. Ensure you have a few pages of order data covering different months and order statuses (tools like WC Smooth Generator can help create sufficient test data).

Testing:

1. Try filtering by month.
    - You should be able to reduce the result set to a single month.
    - You should be able to clear the month filter.
2. Try filtering by customer.
    - You should be able to reduce the result set to those orders placed by a single user.
    - You should be able to clear the customer filter.
    - You should be able to combine the customer filter with the month filter.
3. Ensure status changes via bulk actions still work.
    - Try filtering down to a single month and apply a bulk status change; the filter should remain in effect.

Notes:

- There appears to be a problem with view counts (ie, `All (1) | Completed (1) | ...`) but I believe this is unrelated to this work and should be addressed separately.

### FOR PR REVIEWER ONLY:
